cmd/metal-api/internal/metal/errors.go

```diff
@@ -13,30 +15,68 @@ var (
 
 // NotFound creates a new notfound error with a given error message.
 func NotFound(format string, args ...interface{}) error {
-	return errors.Wrapf(errNotFound, format, args...)
+	return wrapf(errNotFound, format, args...)
 }
 
 // IsNotFound checks if an error is a notfound error.
 func IsNotFound(e error) bool {
-	return errors.Is(errors.Cause(e), errNotFound)
+	return errors.Is(e, errNotFound)
 }
 
 // Conflict creates a new conflict error with a given error message.
 func Conflict(format string, args ...interface{}) error {
-	return errors.Wrapf(errConflict, format, args...)
+	return wrapf(errConflict, format, args...)
 }
 
 // IsConflict checks if an error is a conflict error.
 func IsConflict(e error) bool {
-	return errors.Is(errors.Cause(e), errConflict)
+	return errors.Is(e, errConflict)
 }
 
 // Internal creates a new Internal error with a given error message and the original error.
 func Internal(err error, format string, args ...interface{}) error {
-	return errors.Wrap(errInternal, errors.Wrapf(err, format, args...).Error())
+	return wrapf(errInternal, wrapf(err, format, args...).Error())
 }
 
 // IsInternal checks if an error is a Internal error.
 func IsInternal(e error) bool {
-	return errors.Is(errors.Cause(e), errInternal)
+	return errors.Is(e, errInternal)
+}
+
+// wrapf returns an error annotating err with a stack trace
+// at the point Wrapf is called, and the format specifier.
+// If err is nil, Wrapf returns nil.
+func wrapf(err error, format string, args ...interface{}) error {
```

Comment:
This is the only part which makes me hesitate. Function is obviously taken from https://github.com/pkg/errors/blob/master/errors.go#L201, but is it really an improvement to copy it?

Reply:
Agreed, i will revert this particular change, test are failing because of this i guess

Reply:
reverted